Fix clippy CI error in `coredb-cli`

ianstanton opened the pull request:
It looks like a [clippy error](https://github.com/CoreDB-io/coredb/actions/runs/4813357789/jobs/8569793509) has arisen as we've udpated rust version in CI. This should resolve the issue.

sjmiller609 commented on coredb-cli/tests/integration_tests.rs:
```diff
@@ -18,23 +18,29 @@ fn help() -> Result<(), Box<dyn std::error::Error>> {
     let mut cmd = Command::cargo_bin(CARGO_BIN)?;
     cmd.arg("create")
         .arg("--dry-run")
+        .arg("--resource-type")
```
Is there a way to fix without changing the CLI syntax?

ianstanton commented:
@sjmiller609 it looks like it was a derive bug resolved in a later version of `clap` https://github.com/clap-rs/clap/issues/4733. Removed the previous changes and just updated the crate :)
